sru-templates/manual/exoscale-sru

```diff
@@ -0,0 +1,125 @@
+$exo privnet create $SRU_PRIVNET_NAME --description "A test privnet for use during SRU testing" --netmask 255.255.255.0 --startip 10.0.0.1 --endip 10.0.0.200
+
+# Create a security group, and ensure it allows SSH in
+$exo firewall create $SRU_SECURITY_GROUP
+$exo firewall add $SRU_SECURITY_GROUP ssh
+
+# We pass -i to make sure we are using the key we uploaded to exoscale.
+sshopts=(-i $SSH_KEY -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR )
+
+for series in $SRU_SERIES; do
+
+    case $series in
+        xenial) token="Ubuntu 16.04";;
+        bionic) token="Ubuntu 18.04";;
+        cosmic) token="Ubuntu 18.10";;
+        disco) token="Ubuntu 19.04";;
+        *) echo "Unknown series, update exoscale script!"; exit 1;;
+    esac
+
+    # find the template ID
+    template_id=$($exo vm template list --output-format text | grep $token | cut -f1)
+
+    $exo vm create $series-sru-test -t $template_id -k $SRU_SSH_KEY -p $SRU_PRIVNET_NAME
+
+    VM_IP=$($exo vm list --output-format text | grep $series-sru-test | cut -f6)
+
+    # Wait for the VM to be up
+    while ! ssh "${sshopts[@]}" ubuntu@$VM_IP -- cloud-init status --wait --long; do
+        sleep 5
+    done
+
+    # Capture current state
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- dpkg-query --show cloud-init
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- sudo cat /run/cloud-init/result.json
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- grep Trace /var/log/cloud-init.log
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- systemd-analyze
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- systemd-analyze blame
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- cloud-init analyze show
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- cloud-init analyze blame
+
+    # Upgrade to -proposed cloud-init and reboot
+    scp "${sshopts[@]}" setup_proposed.sh ubuntu@$VM_IP:.
+    ssh "${sshopts[@]}" ubuntu@$VM_IP sudo bash ./setup_proposed.sh 2>&1 | egrep 'cloud-init'
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- dpkg-query --show cloud-init
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- sudo hostname SRU-didnt-work
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- sudo cloud-init clean --logs --reboot
+    ssh-keygen -f "/home/ubuntu/.ssh/known_hosts" -R $VM_IP
+
+    sleep 10  # Wait for the instance to actually go down
+    while ! ssh "${sshopts[@]}" ubuntu@$VM_IP -- cloud-init status --wait --long; do
+        sleep 5
+    done
+
+    echo "Rebooted with the SRU version of cloud-init"
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- hostname
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- "grep Trace /var/log/cloud-init*"
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- systemd-analyze
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- systemd-analyze blame
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- cloud-init analyze show
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- cloud-init analyze blame
+
+    # Check that the private network interface is up - this should show *two* interfaces (3 with lo)
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- ifconfig -s
```

Comment:
This ifconfig isn't always present; I suggest using ``ip`` output, or /sys/class/net/* greping.